nova_vm/src/ecmascript/builtins/keyed_collections/map_objects/map_constructor.rs

```diff
@@ -145,12 +148,33 @@ impl MapConstructor {
     }
 
     fn group_by(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _this_value: Value,
-        _arguments: ArgumentsList,
-        _gc: GcScope,
+        arguments: ArgumentsList,
+        mut gc: GcScope,
     ) -> JsResult<Value> {
-        todo!()
+        let items = arguments.get(0);
+        let callback_fn = arguments.get(1);
+        // 1. Let groups be ? GroupBy(items, callback, collection).
+        let mut groups = group_by_property(agent, items, callback_fn, gc.reborrow())?;
+        // 2. Let map be ! Construct(%Map%).
+        let map = agent.current_realm().intrinsics().map();
+        let map = construct(agent, map.into_function(), None, None, gc.reborrow())?
+            .unbind()
+            .bind(gc.nogc());
+        let map = Map::try_from(map).unwrap();
+
+        // 3. For each Record { [[Key]], [[Elements]] } g of groups, do
+        for g in groups.iter_mut() {
+            // a. Let elements be CreateArrayFromList(g.[[Elements]]).
+            let elements = create_array_from_scoped_list(agent, g.elements.clone(), gc.nogc());
+            // b. Let entry be the Record { [[Key]]: g.[[Key]], [[Value]]: elements }.
+            let _entry = ObjectEntry::new_data_entry(g.key.get(agent), elements.into_value());
+            // c. Append entry to map.[[MapData]].
+            todo!()
```

Comment:
“I don’t know how to implement this part, so I left it as a TODO. 😱😭”

Reply:
This should be basically equivalent to step 6 of `Map.prototype.set`, so you can find the roughly correct code over in `map_prototype.rs:455`.

Reply:
Thank you! Honestly, I’m not confident, but I tried making it while using your reference.
Some tests are still failing, so I think something is probably missing. https://github.com/trynova/nova/pull/559/commits/e50795540d9f718e4d999fd18ff4e5499e1425e5

Reply:
Hmm, from a quick read, this all looks good. I guess figuring out why the tests are failing would need proper investigation :(